primitives/transaction-pool/src/pool.rs

```diff
@@ -167,58 +167,50 @@ pub trait TransactionPool: Send + Sync {
 	// Networking
 
 	/// Returns a future that imports a bunch of unverified transactions to the pool.
-	fn submit_at(
-		&self,
-		at: &BlockId<Self::Block>,
-		xts: impl IntoIterator<Item=TransactionFor<Self>> + 'static,
-	) -> Box<dyn Future<Output=Result<
+	fn submit_at<'a>(
+		&'a self,
+		at: &'a BlockId<Self::Block>,
+		xts: Vec<TransactionFor<Self>>,
+	) -> std::pin::Pin<Box<dyn Future<Output=Result<
 		Vec<Result<TxHash<Self>, Self::Error>>,
 		Self::Error
-	>> + Send + Unpin>;
+	>> + Send + 'a>>;
 
 	/// Returns a future that imports one unverified transaction to the pool.
-	fn submit_one(
-		&self,
-		at: &BlockId<Self::Block>,
+	fn submit_one<'a>(
+		&'a self,
+		at: &'a BlockId<Self::Block>,
 		xt: TransactionFor<Self>,
-	) -> Box<dyn Future<Output=Result<
+	) -> std::pin::Pin<Box<dyn Future<Output=Result<
 		TxHash<Self>,
 		Self::Error
-	>> + Send + Unpin>;
-
-	// RPC
+	>> + Send + 'a>>;
 
+	// *** RPC
 	/// Returns a future that import a single transaction and starts to watch their progress in the pool.
-	fn submit_and_watch(
-		&self,
-		at: &BlockId<Self::Block>,
+	fn submit_and_watch<'a>(
+		&'a self,
+		at: &'a BlockId<Self::Block>,
 		xt: TransactionFor<Self>,
-	) -> Box<dyn Future<Output=Result<Box<TransactionStatusStreamFor<Self>>, Self::Error>> + Send + Unpin>;
-
-
-	// Block production / Networking
+	) -> std::pin::Pin<Box<dyn Future<Output=Result<Box<TransactionStatusStreamFor<Self>>, Self::Error>> + Send + 'a>>;
```

Comment:
Maybe just import `Pin`, cause it's used 3 times here already.

Reply:
> Looks good to me, I think the lifetimes are acceptable.

I haven't also figured out how to patch RPC, probably some more lifetime bounds will be required there 🤷‍♀️